subsys/net/ip/net_context.c

```diff
@@ -2512,25 +2512,38 @@ static int recv_raw(struct net_context *context,
 
 	ARG_UNUSED(timeout);
 
-	context->recv_cb = cb;
-
+	/* If the context already has a connection handler, it means it's
+	 * already registered. In that case, all we have to do is 1) update
+	 * the callback registered in the net_context and 2) update the
+	 * user_data using net_conn_update().
+	 *
+	 * The callback function passed to net_conn_update() must be the same
+	 * function as the one passed to net_conn_register(), not the callback
+	 * set for the net context passed by recv_raw().
+	 */
 	if (context->conn_handler) {
-		net_conn_unregister(context->conn_handler);
-		context->conn_handler = NULL;
-	}
+		context->recv_cb = cb;
+		ret = net_conn_update(context->conn_handler,
+				      net_context_raw_packet_received,
+				      user_data,
+				      NULL, 0);
+	} else {
 
-	ret = bind_default(context);
-	if (ret) {
-		return ret;
-	}
+		ret = bind_default(context);
+		if (ret) {
+			return ret;
+		}
```

Comment:
Same thing here, no need for the else branch, just return after calling `net_conn_update()`

Reply:
Fixed